lab_1/main.py

```diff
@@ -3,87 +3,128 @@
+    return tokens
 
-def remove_stop_words(tokens, stop_words):
-    if isinstance(stop_words, list) and isinstance(tokens, list):
-        tokens_new = []
-        for i in tokens:
-            if i not in stop_words:
-                tokens_new += [i]
-        return tokens_new
-    else:
+
+def remove_stop_words(tokens: list, stop_words: list) -> list or None:
+    """
+    Removes stop words
+    :param tokens: a list of tokens
+    :param stop_words: a list of stop words
+    :return: a list of tokens without stop words
+    """
+    if not (isinstance(stop_words, list) and isinstance(tokens, list)):
         return None
+    tokens_new = []
+    for i in tokens:
+        if i not in stop_words:
+            tokens_new.append(i)
+    return tokens_new
 
 
-def calculate_frequencies(tokens_new):
-    if isinstance(tokens_new, list) and isinstance(tokens_new[0], str):
-        freq_dict = {}
-        for i in tokens_new:
-            if i not in freq_dict:
-                freq_dict[i] = 1
-            else:
-                freq_dict[i] += 1
-        return freq_dict
-    else:
+def calculate_frequencies(tokens_new: list) -> dict or None:
+    """
+   Calculates frequencies of given tokens
+   :param tokens: a list of tokens
+   :return: a dictionary with frequencies
+   """
+    if not (isinstance(tokens_new, list) and isinstance(tokens_new[0], str)):
         return None
+    freq_dict = {}
+    for i in tokens_new:
+        if i not in freq_dict:
+            freq_dict[i] = 1
+        else:
+            freq_dict[i] += 1
+    return freq_dict
 
 
 
-def get_top_n_words(freq_dict, top_n):
-    if isinstance(freq_dict, dict):
-        top_n_words = []
-        if top_n > len(freq_dict):
-            top_n = len(freq_dict)
-        for i in range(top_n):
-            max_value = max(freq_dict.values())
-            for k, v in freq_dict.items():
-                if v == max_value:
-                    max_key = k
-                    break
-            top_n_words.extend([max_key])
-            del freq_dict[max_key]
-        return top_n_words
-    else:
+def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
+    """
+    Returns the most common words
+    :param freq_dict: a dictionary with frequencies
+    :param top_n: a number of the most common words
+    :return: a list of the most common words
+    """
+    if not (isinstance(freq_dict, dict) and isinstance(top_n, int)):
         return None
-
-def create_language_profile(language, text, stop_words):
+    if top_n <= 0:
+        return []
+    top_n_words = []
+    if top_n > len(freq_dict):
+        top_n = len(freq_dict)
+    freq_dict_copy = freq_dict.copy()
+    top_n_copy = top_n
+    while top_n_copy != 0:
+        max_value = max(freq_dict_copy.values())
+        for key, value in freq_dict_copy.items():
+            if value == max_value:
+                max_key = key
+                break
+        top_n_words.append(max_key)
+        del freq_dict_copy[max_key]
```

Comment:
you didn't take my comment about `del` into account...

Reply:
Please, see answer below.